lib/fizzy/execute.cpp

```diff
@@ -574,6 +573,8 @@ ExecutionResult execute(
     const auto& code = instance.module->get_code(func_idx);
     auto* const memory = instance.memory.get();
 
+    const auto local_ctx = ctx.create_local_context();
```

Comment:
Does moving this action from `invoke` to `execute` change the depth that host functions see?

Reply:
No. We record the value in many tests: https://github.com/wasmx/fizzy/blob/faddd97f55ab36cb3ff5214e604707076b0e4f08/test/unittests/execute_call_depth_test.cpp#L65.

Reply:
I thought so, but why?
Before: `call` opcode calls `invoke_function`, it increments depth, calls `execute`, it calls host function.
After:  `call` opcode calls `invoke_function`, calls `execute`, it calls host function. (increment only if it's not imported)

Reply:
Before: `execute(depth: D)`, `call` opcode, `invoke_function()`: increment depth, `execute(depth: D+1)`.
After: `execute(depth: D): increment depth, `call` opcode, `invoke_function()`, `execute(depth: D+1)`.

So this changes the value of the `depth` only inside the `execute()`.

This is needed because later I want to get access the `ExecutionContext` shared stack space in the same place where `depth` is incremented.

I was not able to also move the depth check to the same place easily. This may be handled later, e.g. by trapping inside the `invoke_function()`.